instrumentation/xxl-job/xxl-job-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/xxljob/common/XxlJobCodeAttributesGetter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.xxljob.common;
+
+import static io.opentelemetry.javaagent.instrumentation.xxljob.common.XxlJobConstants.SCRIPT_JOB_TYPE;
+
+import com.xxl.job.core.glue.GlueTypeEnum;
+import io.opentelemetry.instrumentation.api.incubator.semconv.code.CodeAttributesGetter;
+import javax.annotation.Nullable;
+
+public class XxlJobCodeAttributesGetter implements CodeAttributesGetter<XxlJobProcessRequest> {
+
+  @Nullable
+  @Override
+  public Class<?> getCodeClass(XxlJobProcessRequest xxlJobProcessRequest) {
+    GlueTypeEnum glueTypeEnum = xxlJobProcessRequest.getGlueTypeEnum();
+    if (!SCRIPT_JOB_TYPE.contains(glueTypeEnum.getDesc())) {
+      return xxlJobProcessRequest.getDeclaringClass();
+    }
+    return null;
+  }
+
+  @Nullable
+  @Override
+  public String getMethodName(XxlJobProcessRequest xxlJobProcessRequest) {
+    GlueTypeEnum glueTypeEnum = xxlJobProcessRequest.getGlueTypeEnum();
+    if (!SCRIPT_JOB_TYPE.contains(glueTypeEnum.getDesc())) {
+      String methodName = xxlJobProcessRequest.getMethodName();
+      if (methodName == null || methodName.isEmpty()) {
+        return "execute";
+      } else {
+        return methodName;
+      }
+    }
+    return "ID-" + xxlJobProcessRequest.getJobId();
```

Comment:
https://github.com/open-telemetry/semantic-conventions/blob/1e7bb0e4626c2ebb4b3032b8449dc39464d589b7/docs/general/attributes.md?plain=1#L389 does not state that this attribute could be used for anything else besides a method name

Reply:
the description of `code.function` reads `The method or function name, or equivalent (usually rightmost part of the code unit's name).`. It does not mention that this attribute could be set to anything else besides a method name like `"ID-"+jobId`.

Reply:
> the description of `code.function` reads `The method or function name, or equivalent (usually rightmost part of the code unit's name).`. It does not mention that this attribute could be set to anything else besides a method name like `"ID-"+jobId`.

This is a point worth discussing. For some script scheduling job in XXL-JOB, such as `Shell script` or `Python script`, they don't have method:
```
// hello_world.sh
#!/bin/bash
echo "xxl-job:hello shell"
exit 0

# hello_world.py
print("Hello, World!")
```
So I used their's `jobId` field (https://github.com/xuxueli/xxl-job/blob/761de38a0b2a39706e2008e7914fba13bf4ca184/xxl-job-core/src/main/java/com/xxl/job/core/handler/impl/ScriptJobHandler.java#L17) to construct a name to represent method name. It is consistent with the `spanName` I construct in `XxlJobSpanNameExtractor` class. As you mentioned above, it don't contain this situation in semantic convention. Do you have any other alternative way to do it better?

Reply:
> In my opinion the solution is simply not to fill the attributes that you can't fill with appropriate values. I have done this and a bit more in [steverao#6](https://github.com/steverao/opentelemetry-java-instrumentation/pull/6)

Thank you very much! I gained a lot from your reworked code. I also modified some variable names based on your PR. Please review them again when you are free.